lib/Site.js

```diff
@@ -183,10 +187,22 @@ Site.prototype.createPageData = function (config) {
   const sourcePath = path.join(this.rootPath, config.pageSrc);
   const tempPath = path.join(this.tempPath, config.pageSrc);
   const resultPath = path.join(this.outputPath, setExtension(config.pageSrc, '.html'));
+
+  let faviconUrl;
+  if (config.faviconPath) {
+    faviconUrl = url.join(config.baseUrl, config.faviconPath);
+    if (!fs.existsSync(path.join(this.rootPath, config.faviconPath))) {
+      logger.warn(`${config.faviconPath} does not exist`);
+    }
+  } else if (fs.existsSync(path.join(this.rootPath, FAVICON_DEFAULT_PATH))) {
+    faviconUrl = url.join(config.baseUrl, FAVICON_DEFAULT_PATH);
+  }
+
```

Comment:
Let's move this to `Site.prototype.generatePages` since it only needs to be calculated once.
Then pass it in as `config.faviconUrl`.